### ISSUE TITLE
Discussion - change default behavior to assume that regex should be bounded

### ISSUE THREAD

**Issue by wiz0floyd** [ServiceNowDevProgram/SlackerBot#337]
Current behavior: mid-word matches trigger related parsers.

Suggested behavior: implicit /b at the beginning and end for all matching and add a checkbox to the parser to explicitly indicate when mid word matching is intended behavior.

**Comment by eriemer1** [comment]
Makes sense to me. We could also have an optional flag for the parser to explicitly be not bounded when that makes sense, like the @chuck.tomasi one

**Comment by wiz0floyd** [comment]
Alternatively we could have implicit white space on the ends so that mid sentence still triggers but not mid word.
